core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java

```diff
@@ -19,115 +19,299 @@
+    /**
+     * Construct new GeoDistanceQueryBuilder.
+     * @param fieldName name of indexed geo field to operate distance computation on.
+     * */
+    public GeoDistanceQueryBuilder(String fieldName) {
+        Preconditions.checkArgument(Strings.isEmpty(fieldName) == false);
+        this.fieldName = fieldName;
+    }
 
-    public GeoDistanceQueryBuilder(String name) {
-        this.name = name;
+    /** Name of the field this query is operating on. */
+    public String fieldName() {
+        return this.fieldName;
     }
 
-    public GeoDistanceQueryBuilder point(double lat, double lon) {
-        this.lat = lat;
-        this.lon = lon;
+    /** Sets the center point for the query.
+     * @param point the center of the query
+     **/
+    public GeoDistanceQueryBuilder point(GeoPoint point) {
+        this.center = point;
         return this;
     }
 
-    public GeoDistanceQueryBuilder lat(double lat) {
-        this.lat = lat;
+    /**
+     * Sets the center point of the query.
+     * @param lat latitude of center
+     * @param lon longitude of center
+     * */
+    public GeoDistanceQueryBuilder point(double lat, double lon) {
+        this.center = new GeoPoint(lat, lon);
         return this;
     }
 
-    public GeoDistanceQueryBuilder lon(double lon) {
-        this.lon = lon;
-        return this;
+    /** Returns the center point of the distance query. */
+    public GeoPoint point() {
+        return this.center;
     }
 
+    /** Sets the distance from the center using the default distance unit.*/
     public GeoDistanceQueryBuilder distance(String distance) {
-        this.distance = distance;
+        return this.distance(distance, DistanceUnit.DEFAULT);
+    }
+
+    /** Sets the distance from the center for this query. */
+    public GeoDistanceQueryBuilder distance(String distance, DistanceUnit unit) {
+        Preconditions.checkArgument(Strings.isEmpty(fieldName) == false);
+        this.distance = DistanceUnit.parse(distance, unit, DistanceUnit.DEFAULT);
         return this;
     }
 
+    /** Sets the distance from the center for this query. */
     public GeoDistanceQueryBuilder distance(double distance, DistanceUnit unit) {
-        this.distance = unit.toString(distance);
+        this.distance = DistanceUnit.DEFAULT.convert(distance, unit);
         return this;
     }
 
+    /** Returns the distance configured as radius. */
+    public double distance() {
+        return distance;
+    }
+
+    /** Sets the center point for this query. */
     public GeoDistanceQueryBuilder geohash(String geohash) {
-        this.geohash = geohash;
+        this.center.resetFromGeoHash(geohash);
         return this;
     }
 
+    /** Which type of geo distance calculation method to use. */
     public GeoDistanceQueryBuilder geoDistance(GeoDistance geoDistance) {
-        this.geoDistance = geoDistance;
+        this.geoDistance = (geoDistance != null) ? geoDistance : DEFAULT_GEO_DISTANCE;
```

Comment:
Wrt. to recent discussions we had: Should we change that to barf instead of silently dealing with the null value?